packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -2089,15 +1311,16 @@ describe('NetworkController', () => {
           },
         },
         async ({ controller }) => {
-          const fakeMetamaskProvider = buildFakeMetamaskProvider();
-          createMetamaskProviderMock.mockReturnValue(fakeMetamaskProvider);
+          const fakeProvider = buildFakeProvider();
+          const fakeNetworkClient = buildFakeNetworkClient(fakeProvider);
+          createNetworkClientMock.mockReturnValue(fakeNetworkClient);
 
           await controller.setActiveNetwork('testNetworkConfigurationId');
 
           expect(controller.state.providerConfig).toStrictEqual({
             type: 'rpc',
             rpcTarget: 'https://mock-rpc-url',
-            chainId: '0xtest',
```

Comment:
This was never a valid chain ID, but was okay prior to this PR because we didn't try to convert the `chainId` to a hex string when building the provider. Now we do (see `#configureProvider`).